### PR TITLE
Mac address endianness fix

### PR DESCRIPTION
**Pull request by komaljai**
"dstAddr" and "srcAddr" fields are of type "macaddr", which is stored in sk_buff in little endian form. 
load_dword will convert big_endian to little_endian and load it to cpu. The issue is ebpf compiler shouldn't call load_dword when there is annotation 'macaddr', instead just load the little-endian format to cpu.

### CODE REVIEW

File: backends/ebpf/runtime/ebpf_kernel.h [hunk 1]
```diff
@@ -41,6 +41,7 @@ limitations under the License.
 #define load_half(data, b) bpf_ntohs(*(u16 *)((u8*)(data) + (b)))
 #define load_word(data, b) bpf_ntohl(*(u32 *)((u8*)(data) + (b)))
 #define load_dword(data, b) bpf_be64_to_cpu(*(u64 *)((u8*)(data) + (b)))
+#define swap_endianess(x) ___bpf_swab64(x)
```
Comment (usha1830) [inline backends/ebpf/runtime/ebpf_kernel.h]: Where can I find the list of these builtin functions

File: backends/tc/ebpfCodeGen.cpp [hunk 2]
```diff
@@ -408,13 +408,173 @@ void TCIngressPipelinePNA::emitLocalVariables(EBPF::CodeBuilder *builder) {
...
+                checkIfMAC = true;
+                break;
+            }
+        }
+    }
+
+    msgStr = Util::printf_format("Parser: extracting field %s", fieldName);
+    builder->target->emitTraceMessage(builder, msgStr.c_str());
+
+    if (widthToExtract <= 64) {
```
Comment (usha1830) [inline backends/tc/ebpfCodeGen.cpp:448]: We need to update the code to use memcpy for macaddr. Perhaps a lot of code will go away from this

**Review by usha1830 — COMMENTED** [review]
Please check comments

File: backends/ebpf/runtime/ebpf_kernel.h (hunk 1 above)
Comment (komaljai) [reply backends/ebpf/runtime/ebpf_kernel.h]: /p4c/backends/ebpf/runtime/contrib/libbpf/src/bpf_endian.h

File: backends/tc/ebpfCodeGen.cpp (hunk 2 above)
Comment (komaljai) [reply backends/tc/ebpfCodeGen.cpp:448]: resolved

**Pull request closed without merging**